src/main/java/io/github/mountainrange/mule/gameplay/WorldMap.java

```diff
@@ -30,37 +36,72 @@ public WorldMap(Grid<VisualTile> g, MapType mType) {
 		}
 	}
 
-	public WorldMap(Grid g, MapType mType, List<Player> playerList) {
+	public WorldMap(Grid<VisualTile> g, MapType mType, List<Player> playerList) {
 		this(g, mType);
 
-		ownedTiles = new HashMap<>();
+		productionTiles = new HashMap<>();
 		for (Player p : playerList) {
-			ownedTiles.put(p, new HashSet<>());
+			productionTiles.put(p, new EnumMap<>(MuleType.class));
 		}
 	}
 
 	// ----------------------------Logical methods-------------------------------
 
 	/**
-	 * Count the number of grid owned by the given player, or zero if the given player owns no grid.
-	 * @param player player to count owned grid
+	 * Count the number of grid owned by the given player, or zero if the given player owns no tiles.
+	 * @param player player to count owned tiles
 	 * @return number of grid owned by the given player
 	 */
 	public int countLandOwnedBy(Player player) {
-		if (!ownedTiles.containsKey(player)) {
+		if (!productionTiles.containsKey(player)) {
 			return 0;
 		}
-		return ownedTiles.get(player).size();
+
+		// Count all the tiles in the individual Sets owned by the given player
+		return productionTiles.get(player).values().stream().mapToInt(Set::size).sum();
 	}
 
 	/**
 	 * Return an unmodifiable set with the grid owned by the given player, or an empty set if the given player owns no
-	 * grid.
-	 * @param player player to get owned grid
+	 * tiles.
+	 * @param player player to get owned tiles
 	 * @return unmodifiable set with grid owned by the given player
 	 */
 	public Set<Tile> landOwnedBy(Player player) {
-		return Collections.unmodifiableSet(ownedTiles.getOrDefault(player, new HashSet<>()));
+		if (!productionTiles.containsKey(player)) {
+			return EMPTY_SET;
+		}
+
+		// Construct a flat Set of all the Tiles a given player owns
+		return productionTiles.get(player).values().stream()
+				.flatMap(Set::stream)
+				.collect(Collectors.toCollection(HashSet::new));
+	}
+
+	/**
+	 * Count the number of tiles owned by the given player with the given MULE installed.
+	 * @param player player who owns the tiles
+	 * @param mule mule installed on the tiles
+	 * @return number of tiles that the player owns with the mule installed
+	 */
+	public int countTilesWithMule(Player player, MuleType mule) {
+		if (!productionTiles.containsKey(player)) {
+			return 0;
+		}
+		return productionTiles.get(player).get(mule).size();
+	}
+
+	/**
+	 * Return an unmodifiable set with all the tiles a player owns with the given mule installed.
+	 * @param player player who owns the tiles
+	 * @param mule mule installed on the tiles
+	 * @return unmodifiable set of tiles with the given mule installed and owned by the given player
+	 */
+	public Set<Tile> tilesWithMule(Player player, MuleType mule) {
+		if (!productionTiles.containsKey(player)) {
+			return EMPTY_SET;
+		}
+		return Collections.unmodifiableSet(productionTiles.get(player).get(mule));
```

Comment:
This was a good idea, we should apply unmodifiable<stuff> to more places later on. That might break some things though...